src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -456,15 +456,18 @@ public function makePathRelative($endPath, $startPath)
         }
 
         $stripDriveLetter = function ($path) {
-            if (\strlen($path) > 2 && ':' === $path[1] && '/' === $path[2] && ctype_alpha($path[0])) {
-                return substr($path, 2);
-            }
-
-            return $path;
+            return (\strlen($path) > 2 && ':' === $path[1] && '/' === $path[2] && ctype_alpha($path[0]))
+                ? [substr($path, 2), $path[0]]
+                : [$path, null];
         };
 
-        $endPath = $stripDriveLetter($endPath);
-        $startPath = $stripDriveLetter($startPath);
+        list($endPath, $endDriveLetter) = $stripDriveLetter($endPath);
+        list($startPath, $startDriveLetter) = $stripDriveLetter($startPath);
+
+        if ($endDriveLetter && $startDriveLetter && $endDriveLetter != $startDriveLetter) {
+            // Different drives
+            return $endDriveLetter.':'.$endPath;
```

Comment:
what if the end path contains `..`? don't we want to resolve these?

Reply:
I think we do.

Changed, so that path is normalised: 
`makePathRelative('D:/../../aa/../bb/cc', 'C:/aa/dd/..')` → `D:/bb/cc/`

Also, difference in letter case is now properly handled: 
`makePathRelative('C:/aa/bb/cc', 'c:/aa/dd/..')` → `bb/cc/`